### ISSUE TITLE
nerdctl can not create volume successfully

### ISSUE THREAD

**Issue by Huimintai** [containerd/nerdctl#262]
I want to use `-v` command to create a volume and mount into container path:
```
docker run -d --name registry-https --restart always -p 443:443 -v /opt/tke-installer/registry:/var/lib/registry -v registry-certs:/certs -e REGISTRY_HTTP_ADDR=0.0.0.0:443 -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/server.crt -e REGISTRY_HTTP_TLS_KEY=/certs/server.key registry.tke.com/library/registry-amd64:2.7.1
```
For `docker run`command it can create the `registry-certs`and I can check certs  from `/varlib/docker/volume/registry-certs/_data`
```
# ls /var/lib/docker/volumes/registry-certs/_data/
server.crt  server.key
```
But for nerdctl it can not create the `registry-certs` volume:
```
# nerdctl run -d --name registry-https-test --restart always -p 443:443 -v /opt/tke-installer/registry:/var/lib/registry -v registry-certs:/certs -e REGISTRY_HTTP_ADDR=0.0.0.0:443 -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/server.crt -e REGISTRY_HTTP_TLS_KEY=/certs/server.key registry.tke.com/library/registry-amd64:2.7.1
FATA[0000] volume "registry-certs" not found: not found
```
Does `nerdctl`can not support create volume when `nerdctl run -v`?
How to use `nerdctl -v`command to create volume like docker?
Thanks a lot.

**Comment by fahedouch** [comment]
Hi @Huimintai ,

the volume should start with `/` (e.g `registry-certs`) otherwise it will treat it like a nerdctl volume

**Comment by Huimintai** [comment]
> Hi @Huimintai ,
> 
> the volume should start with `/` (e.g `/registry-certs`) otherwise it will treat it like a nerdctl volume
> 
> sthg like this :
> 
> ```
> # nerdctl run -d --name registry-https-test --restart always -p 443:443 -v /opt/tke-installer/registry:/var/lib/registry -v /registry-certs:/certs -e REGISTRY_HTTP_ADDR=0.0.0.0:443 -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/server.crt -e REGISTRY_HTTP_TLS_KEY=/certs/server.key registry.tke.com/library/registry-amd64:2.7.1
> ```

The registry-certs is a nerdctl volume.I want to nerdctl can create the volume for me.It is not a path.
The docker command can create the `registry-certs`for me although I didn't create the volume before run the `docker run -v registry-certs:/certs `.
```
# docker volume ls
DRIVER              VOLUME NAME
local               registry-certs
```
# ls /var/lib/docker/volumes/registry-certs/_data/
server.crt  server.key
```

```

**Comment by AkihiroSuda** [comment]
Currently it has to be created with `nerdctl volume create <NAME>`

**Comment by Huimintai** [comment]
Yes,I can create the volume manully then I mount the volume into docker container.But I can not find the data from the volume:
```
# nerdctl volume create registry-certs
registry-certs
```
```
# nerdctl run -d --name registry-https-test --restart always -p 443:443 -v /opt/tke-installer/registry:/var/lib/registry -v registry-certs:/certs -e REGISTRY_HTTP_ADDR=0.0.0.0:443 -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/server.crt -e REGISTRY_HTTP_TLS_KEY=/certs/server.key registry.tke.com/library/registry-amd64:2.7.1
7fdc4dc42221ea59373bf5a98cbff261f5d6d6e76933497c85f88cc90289c00d
```
```
# nerdctl volume ls
VOLUME NAME       DIRECTORY
registry-certs    /var/lib/nerdctl/1935db59/volumes/default/registry-certs/_data
root@VM-0-28-ubuntu:~# ls  /var/lib/nerdctl/1935db59/volumes/default/registry-certs/_data
root@VM-0-28-ubuntu:~#
```
You can see it is empty without my data.But for docker container I can see the data :
```
# ls /var/lib/docker/volumes/registry-certs/_data/
server.crt  server.key
```

**Comment by AkihiroSuda** [comment]
What's your nerdctl version? Could you try the latest version (v0.9.0)?

**Comment by Huimintai** [comment]
My nerdctl version:
```
# nerdctl version
Client:
 Version:	v0.8.3
 Git commit:	487f94aa7a8cfa833a1e96600d11b0c2bdc931f3

Server:
 containerd:
  Version:	v1.5.2
  Revision:	36cc874494a56a253cd181a1a685b44b58a2e34a
```

**Comment by Huimintai** [comment]
I try to use 0.9.0 to test.It can also can not find the certs in volume.
```
# nerdctl version
Client:
 Version:	v0.9.0
 Git commit:	39a46444754c581e020fffa63261f2a859c16bac

Server:
 containerd:
  Version:	v1.5.2
  Revision:	36cc874494a56a253cd181a1a685b44b58a2e34a
```
```
# docker run -it --rm --entrypoint=sh tkestack/tke-installer-amd64:2107f202f84d0f520983ab741501770a7e1c08d5
/app # ls
assets                certs                 data                  images.tar.gz         provider
bin                   conf                  hooks                 manifests             public.charts.tar.gz
/app # ls bin/
kubectl        tke-installer
/app # ls certs/
server.crt  server.key
/app #
```
```
# nerdctl run --name tke-installer -d --privileged --net=host --restart=always -v /etc/hosts:/app/hosts -v /opt/tke-installer/data:/app/data -v /opt/tke-installer/conf:/app/conf -v registry-certs:/app/certs tkestack/tke-installer-amd64:2107f202f84d0f520983ab741501770a7e1c08d5
24f166302ffd313d412c94b85fd40f7434a7f7bd52b18a7edb2019261cdc04fe
root@VM-0-28-ubuntu:~#
root@VM-0-28-ubuntu:~# nerdctl volume ls
VOLUME NAME                                                         DIRECTORY
4cb05dd0fbe7cdf6909f183066aa0da60dcf3900d8d4c680a499eb2dc62ec36d    /var/lib/nerdctl/1935db59/volumes/default/4cb05dd0fbe7cdf6909f183066aa0da60dcf3900d8d4c680a499eb2dc62ec36d/_data
registry-certs                                                      /var/lib/nerdctl/1935db59/volumes/default/registry-certs/_data
test                                                                /var/lib/nerdctl/1935db59/volumes/default/test/_data
tke-installer-bin                                                   /var/lib/nerdctl/1935db59/volumes/default/tke-installer-bin/_data
root@VM-0-28-ubuntu:~# ls /var/lib/nerdctl/1935db59/volumes/default/registry-certs/_data
root@VM-0-28-ubuntu:~#
```

**Comment by fahedouch** [comment]
@AkihiroSuda  https://github.com/containerd/nerdctl/blob/39a46444754c581e020fffa63261f2a859c16bac/run_mount.go#L110 here `imageVolumes`  seems to be empty. it detect only imageVolumes that are defined in Dockerfile "VOLUME" instruction. I takz care about the fix

**Comment by AkihiroSuda** [comment]
thanks!

**Comment by Huimintai** [comment]
@AkihiroSuda @fahedouch Thanks for all your quick response and fix:>>

**Comment by AkihiroSuda** [comment]
Yes, it should be fixed in #263.
Will be included in v0.10.0.

**Comment by Huimintai** [comment]
Thanks for your hardworks.Nerdctl can run successfully:>
```
# nerdctl run  -d --privileged --net=host --restart=always -v /etc/hosts:/app/hosts -v /run/containerd/containerd.sock:/run/containerd/containerd.sock -v /opt/tke-installer/data:/app/data -v /opt/tke-installer/conf:/app/conf -v registry-certs:/app/certs -v tke-installer-bin:/app/bin docker.io/tkestack/tke-installer-amd64:2107f202f84d0f520983ab741501770a7e1c08d5 tke-installer
1ebeaf9c038ae07ea762b6717848236136f719a7b83242f4577995503bc2ba6f
root@VM-8-15-ubuntu:~#
root@VM-8-15-ubuntu:~#
root@VM-8-15-ubuntu:~# nerdctl ps
CONTAINER ID    IMAGE                                                                              COMMAND                   CREATED          STATUS    PORTS    NAMES
bc7399322ae1    docker.io/tkestack/tke-installer-amd64:2107f202f84d0f520983ab741501770a7e1c08d5    "/app/bin/tke-instal…"    5 minutes ago    Up
```
```
# nerdctl volume ls
VOLUME NAME          DIRECTORY
registry-certs       /var/lib/nerdctl/1935db59/volumes/default/registry-certs/_data
tke-installer-bin    /var/lib/nerdctl/1935db59/volumes/default/tke-installer-bin/_data
```
